sklearn/ensemble/gradient_boosting.py

```diff
@@ -1712,14 +1712,19 @@ def feature_importances_(self):
         """
         self._check_initialized()
 
-        total_sum = np.zeros((self.n_features_, ), dtype=np.float64)
-        for stage in self.estimators_:
-            stage_sum = sum(tree.tree_.compute_feature_importances(
-                normalize=False) for tree in stage) / len(stage)
-            total_sum += stage_sum
-
-        importances = total_sum / total_sum.sum()
-        return importances
+        relevant_trees = [tree
+                          for stage in self.estimators_ for tree in stage
+                          if tree.tree_.node_count > 1]
+        if not relevant_trees:
+            # degenerate case where all trees have only one node
+            return np.zeros(shape=self.n_features_)
```

Comment:
We can test this by growing single node trees:

```python
x = np.zeros((10, 10))
y = np.ones((10,))
gbr = GradientBoostingRegressor()
gbr.fit(X, y)
```

Do we want this degenerate case to also sum to one?

Reply:
I don't think it makes sense to return non-zero values for the degenerate case. At least in my mind, the zero vector is a better representative of what we have. It's like, are features equally important, or are they not important at all? I think they're not important at all in this case.

Reply:
If this does not sum to one, we should document it.

Reply:
added to the docstring